### ISSUE TITLE
Weird flows for other datasets

### ISSUE THREAD

**Issue by HamidGadirov** [hzwer/ECCV2022-RIFE#240]
Hi,

Thanks you for developing interesting method, I am planning to use it in my research. I am trying inference with other datasets, even simulation datasets containing actual flow. However, in some cases the intermediate flow looks weird:
<img width="896" alt="arrow" src="https://user-images.githubusercontent.com/39424882/158145538-f7b3cb2e-bdde-40c8-9e64-98d5ea6d3708.png">
This is flow encoded with arrows (i.e. arrows representing magnitude of the flow). And this is the flow in HSV:
<img width="853" alt="hsv" src="https://user-images.githubusercontent.com/39424882/158145584-94d78701-9fac-442c-8a1c-5b4771cb3c1f.png">
There is a lot of flow for empty regions with zero pixel values where nothing is moving (top left and bottom right parts). What could cause such weird flow results? Interpolation still works fine.

Regards,
Hamid

**Comment by hzwer** [comment]
If you just want accurate optical flow, it is recommended to try RAFT: https://github.com/princeton-vl/RAFT
Maybe I can add constraints to static scenes during RIFE training.

**Comment by HamidGadirov** [comment]
Thanks for your reply, I am aware of RAFT, I was looking at it in the beginning, however the idea of my project is to have unsupervised method because we don't always have the flow labels to train the network. I am planning also to implement RIFE in 3D to support volumetric interpolation.

I tried another scientific dataset, the flow there is meaningful (and interpolation is perfect):
<img width="553" alt="arrows_drop" src="https://user-images.githubusercontent.com/39424882/158179781-c77bf806-e7e7-434a-9bfb-2d70d86bc52a.png">

It is just a totally different case for first dataset that I showed, it looks like something is off there. Could that be related to the fact that data was not in RGB initially? Also, the output size of the flow wasn't same as input frames size, I padded the input, not sure if that is the best thing to do.

**Comment by hzwer** [comment]
You have an exciting news. My suggestion is to make some improvements to the loss function as I don't know if methods like laplacian loss will work in your scenario.

**Comment by HamidGadirov** [comment]
Thanks for your suggestion. I changed the loss from laplacian to l1. I'll train the model with this loss and check the results.

**Comment by HamidGadirov** [comment]
I have modified the loss and removed laplacian pyramid. I trained the model, but when I wanted to check inference, I am getting the error: 
```
"Missing key(s) in state_dict: "block0.conv0.0.weight", "block0.conv0.1.weight", "block0.conv0.1.b
ias", "block0.conv0.1.running_mean", "block0.conv0.1.running_var", "block0.conv0.2.weight", "block0.res0.
conv1.0.weight", "block0.res0.conv1.1.weight", "block0.res0.conv1.1.bias", "block0.res0.conv1.1.running_m
ean", "block0.res0.conv1.1.running_var", "block0.res0.conv1.2.weight", ...
Unexpected key(s) in state_dict: "block_tea.conv0.0.0.weight", "block_tea.conv0.0.0.bias", "block
_tea.conv0.0.1.weight", "block_tea.conv0.1.0.weight", "block_tea.conv0.1.0.bias", "block_tea.conv0.1.1.we
ight", "block_tea.convblock.0.0.weight", "block_tea.convblock.0.0.bias", ..."
```
I am getting a similar error when I train the unmodified model on Vimeo90K and run inference. Am I missing something?

**Comment by HamidGadirov** [comment]
I have fixed the previous issue via just using the training and inference of the same model (RIFE.py, IFNet.py).
And about the weird flow, I think it might be because the IFNet layers are not regularized. I am going to try adding small regularization (L1 and L2) and check the results.
